libutils/include/utils/time.h

```diff
@@ -6,31 +6,72 @@
 
 #pragma once
 
+/*
+ * Time should use a 64-bit data type, because a 32-bit type can overflow too
+ * easily for longer timeouts. However, there are no hard-coded assumptions
+ * about this in here. Postfixing values with "lu" or "llu" is configurable to
+ * allow aligning this with the runtme library definitions for uint64_t and
+ * PRIu64. The macro SEL4_TIME_UINT_TYPE() defaults to UINT64_C(), which is a
+ * common way runtime libraries defines 64-bit datatypes:
+ *
+ *     #if defined(target_is_32_bit_with_LLP64)
+ *         typedef unsigned long long int   uint64_t
+ *         #define UINT64_C(v)              v ## ull
+ *         #define PRIu64                   "ull"
+ *
+ *     #elif defined(target_is_64_bit_with_LP64)
+ *         typedef unsigned long int        uint64_t
+ *         #define UINT64_C(v)              v ## ul
+ *         #define PRIu64                   "ul"
+ *     ...
+ *     #endif
+ *
+ * See also https://en.wikipedia.org/wiki/64-bit_computing for commonly used
+ * 64-bit data models. The seL4 kernel uses "long long" for 64-bit types, see
+ * file <sel4-kernel>/include/stdint.h. The userspace is not required to follow
+ * this, musllibc uses "long long" on 32-bit targets and "long" on 64-bit
+ * targets.
+ */
+
+#ifndef SEL4_TIME_UINT_TYPE
+#define SEL4_TIME_UINT_TYPE(v)  UINT64_C(v)
```

Comment:
I think it would be helpful to add  `#include <stdint.h>` to avoid situations where `UINT64_C` remains undeclared?

Reply:
I'm happy to add it here. I was not sure if the `UINT64_C`macro is guaranteed to be define there and it's best in the end to leave this file "floating" purposely, so we don't impose anything. But practically, `stdint.h` seem to provide this in many libs.

Reply:
It seems to be expected in stdint.h in C99 at least (https://en.cppreference.com/w/c/types/integer)

Reply:
ok, will add the include then.

Reply:
"#include <stdint.h>" added now